pringles/models/models.py

```diff
@@ -34,10 +34,14 @@ def with_output_port(self, name: str) -> AtomicModelBuilder:
         return self
 
     def build(self) -> Any:
-        return type(self.name, (Atomic,), {
+        created_class = type(self.name, (Atomic,), {
             DISCOVERED_INPUT_PORTS_FIELD: self.discovered_input_ports_fields,
             DISCOVERED_OUTPUT_PORTS_FIELD: self.discovered_output_ports_fields
         })
+        # The class is added to the global namespace
+        # This is a workarround for the simulation pickler to find the class
+        globals()[self.name] = created_class
```

Comment:
What happens if there's already some object defined in the global namespace with that name. Say for example, some user imported something like
```python
from falopa import Queue
```
and then a model named `Queue` is being discovered

I'll throw some exception in this case, and add a test case.